src/libs/actions/Session/index.ts

```diff
@@ -763,8 +775,11 @@ function invalidateCredentials() {
 }
 
 function invalidateAuthToken() {
-    NetworkStore.setAuthToken('pizza');
-    Onyx.merge(ONYXKEYS.SESSION, {authToken: 'pizza'});
+    // expires the session after 50s
+    setTimeout(() => {
```

Comment:
Why timeout here? I dont think we need it

Reply:
yes it's necessary, otherwise we barely can observe the reload of images in the tests (like in your own Android:mweb video btw). Not everybody is as quick in clicking as you are :smile:

Reply:
i cannot do the tests without setting the timeout and i'm sure many couldnt. You are doing test2 only in your android mweb video, you cannot do test3 if there is no timeout. Please include all 3 tests in your videos and you could see that timeout is necessary.  We could maybe reduce the lapse time from 50s to 30s.

Reply:
Nope. We don't need timeout, we have a shortcut to open TestTool menu (command + D on web and 4 finger gesture on native).

Reply:
What is the shorcut on native ? Still i cannot do the tests without timeout, i use a VM and i cannot be "running" to show the reloads.  I wish i could be done but it cannot be done to show the reloads properly.

Reply:
https://github.com/Expensify/App/pull/53407#issuecomment-2557408616

Reply:
please upload a video of test 3 on a native platform without timeout, we can know then that you did all the 3 tests and you can do it without timeout

Reply:
Herre you are

https://github.com/user-attachments/assets/af6deeaf-a417-4833-9767-29c766454eae

Reply:
ok, that's step2 but i guess same can be done for step3. I'll test it.

Reply:
@hungvu193 the test preferences are not displayed when on the carousel (test 3). Have you tried ?

Reply:
@hungvu193 it seems we cannot access the test prefences menu from the carousel. I suggess to set the timeout to 10s if you prefer

Reply:
Can we just invalid the session then click the image instead? That's an edge case, no need to setTimeout

Reply:
> Can we just invalid the session then click the image instead? That's an edge case, no need to setTimeout

that will not be a test case of any interest for this PR imho.

I dont know why you are so against a timeout. The thing is that the point of the test is to be able to see how the app reacts to an expired session so a useEffect or a timeout seems the best for that imho.

But another possibility is to set the creation to 2H minus let say 5s, then the tester can come back in the chat and reload the chat 5s later (test 2) or click the image to display the carousel (test 3) but there is no guarantee all this fits perfectly with the 5s for a correct testing experience. Using a timeout seems to be a better solution

Reply:
@hungvu193 i was wondering if we could use the "import onyx state" in the tools. Do you know that tool well ?

Reply:
>  i was wondering if we could use the "import onyx state" in the tools. Do you know that tool well ? We could then import a session expiring in the next 5s and follow the steps in my previous comment. We wont have to use the "invalidate session" tool then.

I don't think we have it, but it's nice to have.

Reply:
I still feel that we don't need to setTimeout or even set `creationDate` to "2Hours - 5 sec" to achieve the testcase.

The expired session on a carousel inside native platforms feels like an edge case to me. If you disagree then I can pull internal engineer here for the final decision. Wdyt?

Reply:
test 3 is important as we have done a lot of work related to the carousel. I suggess we restrict it to the web explaining that we dont have access to the testtool from the carousel but it was tested in our videos using some timeout OR we just let test3 down

Reply:
Ok, let's do it and wrap this PR up.

Reply:
but having a tester adivce or the internal engineer advice is not bad neither, it's useful to let them know of the choice that should be made

Reply:
Ok, Let's remove timeout, I'll approve the PR. Internal Engineer will be requested for review and we will continue discussing about it? How about that?

Reply:
> Ok, let's do it and wrap this PR up.

restrict or remove ?

Reply:
Remove it.

Reply:
Oh sorry, Let's restrict and let's the QA know about the 3 tests.

Reply:
ok, i will remove test3 and redo my videos also then

Reply:
ok, i will restrict test3 then and remove the timeout. But i will let my videos to show that we have tested it on native platforms using a timeout, right ?

Reply:
Yeah. Correct

Reply:
is there a way to let QA know now ? even before delivering the PR

Reply:
@Kalydosos You can leave a comment under QA section to let them know

Reply:
i feel uneasy about this @hungvu193  sorry. I feel like QA will never agree not to be able to test it themselves. Let's talk this with an internal engineer please, sorry for backtracking

Reply:
or let's remove test3 in fine, as formally they cannot test it

Reply:
i think it's better to remove test3

Reply:
Ok let remove it

Reply:
ok, :+1:

Reply:
sorry @hungvu193 i just realize that test3 is the core test of our ticket 51888 and test1 and test2 are the core tests of the other ticket 51699; so we cannot deliver the PR without these tests. It seems like timeout is our only way out

Reply:
cc @rlinoz about the about discussion.

Reply:
I agree we should not add the timeout, that will worsen the development/testing experience for every other case

Also, I don't think validating the fix means necessarily being able to show the loading spinner.

If taking the same actions caused a CORS error before and now it doesn't that should be enough, and maybe we should validate the test by looking at the CORS error

Reply:
@rlinoz in fact we may already have an impact in the development/testing experience of other case. Our core problem is that we no way to test the fix on native platforms. So we change the usage of the test tool "invalidate session" to not only invalidate the session in the BE (as before) but also expires the session in the FE. We dont know if this could be  problem at some point for those who usually use that tool. 

Setting a timeout is a plus because to cause the cors error, the tester need to click on the image when the session is expired. So i see no other way to test the cors error in that case, @hungvu193 do you ?

The perfect solution for native platform will be to create our own test tool but maybe it will be of no use except for this ticket. Will that be ok ?

Reply:
I am fine if we add a new tool like `Invalidate session delayed` as the last item in the `TestToolMenu`, if it is useless after this PR we can remove it later.

Reply:
@rlinoz it's all good then, thx for the help. I will implement that asap :+1:

Reply:
@hungvu193 we will need these 2 translations in Spanish. The process should go through slack right ?

> expiredSession: 'Expire session',
   expireSessionWithDelay: 'Expire with delay'

Reply:
What do you mean by `Send expired session`?

Reply:
@rlinoz  we litteraly send an expired session to the FE

Reply:
We create a new tool called `Invalidate session delayed` not `Send expired session`. Can you update please?

Reply:
@hungvu193 "Invalidate session delayed" is a suggession by @rlinoz but "send expired session" is a more accurate title for what we do in fact. We dont really "invalidate" the session we expires it.

Reply:
@hungvu193 should we ask for the translations ?

Reply:
Yes

Reply:
Please create a slack post on #expensify-open-source to ask for translations.

Reply:
@hungvu193 i was about to change the title of the tool to "Invalidate session delayed" as you seems to disagree. Can you confirm if i still change, i dont think the internatl engineer disagrees as he did mentionned. Please let me know, so we can make sure of what to translate. 

I never ask for translations on Slack before. Can you please create the post so i can see how it is supposed to be asked and who to target ?

Reply:
Ok.
Btw, here is an example slack post that you can refer:

https://expensify.enterprise.slack.com/archives/C01GTK53T8Q/p1734369232859489

Reply:
@hungvu193 ok thx, i create a new post on slack from the link you recommended, here it is https://expensify.slack.com/archives/C01GTK53T8Q/p1735573736570229 , i created one earlier today https://expensify.slack.com/archives/C01GTK53T8Q/p1735558564704039 but we have no answers yet. But this is an internal tool so i dont know if we should wait too long for the translations validation because it really has no impact on the UX or the fix



> Ok. 

By ok you are meaning we keep "Send expired session" as the tool name ? Otherwise if we must change it what do you also recommend for the button label ? Currenly it's "Send with delay"

Reply:
Replied in slack